### PR TITLE
[TH2-1150] Fix filter on MessageGroupBatch router

### PR DESCRIPTION
**Pull request by ivandruzhinin**
* filter MessageGroup's atomically

### CODE REVIEW

**Comment by Ilya-Loginov-Exactpro** [comment]
We will filter only fields in parsed messages? What happened if users want to filter only by metadata?

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchRouter.kt [hunk 1]
```diff
@@ -19,20 +19,44 @@ package com.exactpro.th2.common.schema.message.impl.rabbitmq.group
...
 import com.exactpro.th2.common.schema.message.QueueAttribute.PUBLISH
 import com.exactpro.th2.common.schema.message.QueueAttribute.SUBSCRIBE
 import com.exactpro.th2.common.schema.message.configuration.QueueConfiguration
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.ConnectionManager
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.router.AbstractRabbitBatchMessageRouter
+import com.exactpro.th2.common.schema.strategy.fieldExtraction.impl.AnyMessageFieldExtractionStrategy
+
+class RabbitMessageGroupBatchRouter : AbstractRabbitBatchMessageRouter<MessageGroup, MessageGroupBatch, Builder>() {
+    init {
+        setFilterStrategy(DefaultFilterStrategy(AnyMessageFieldExtractionStrategy()))
```
Comment (Ilya-Loginov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchRouter.kt:33]: We will filter only parsed messages? What happened if users want to filter only by metadata in raw messages?

File: src/main/kotlin/com/exactpro/th2/common/schema/strategy/fieldExtraction/impl/AnyMessageFieldExtractionStrategy.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.schema.message.toJson
+import com.exactpro.th2.common.schema.strategy.fieldExtraction.AbstractTh2MsgFieldExtraction
+
+class AnyMessageFieldExtractionStrategy : AbstractTh2MsgFieldExtraction() {
+    override fun parseMessage(message: com.google.protobuf.Message): Message {
+        check(message is AnyMessage) { "Message is not an ${AnyMessage::class.qualifiedName}: ${message.toJson()}" }
+        check(message.hasMessage()) { "Message doesn't contain a parsed message inside: ${message.toJson()}" }
```
Comment (Ilya-Loginov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/common/schema/strategy/fieldExtraction/impl/AnyMessageFieldExtractionStrategy.kt:27]: We will filter only parsed messages? What happened if users want to filter only by metadata in raw messages? Maybe we should return created message only with metadata?

Comment (ivandruzhinin) [reply src/main/kotlin/com/exactpro/th2/common/schema/strategy/fieldExtraction/impl/AnyMessageFieldExtractionStrategy.kt:27]: At the moment there's no filtering for raw messages, even in existing router for raw message batches. Such filtering would be implemented in task TH2-1503

**Review by Ilya-Loginov-Exactpro — APPROVED** [review]

**Pull request merged**